### PR TITLE
Gcw 2421 add total request count

### PR DESCRIPTION
**Pull request by abulsayyad123**
Hi Team,
This PR is regarding jira ticket GCW-2421- add total request count on summary tab(https://jira.crossroads.org.hk/browse/GCW-2421). Please review the PR and let me know if any change is required. 

Thanks.

### CODE REVIEW

File: app/serializers/api/v1/order_serializer.rb [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module Api::V1
   class OrderSerializer < OrderShallowSerializer
-    attributes :item_ids, :cancellation_reason, :unread_messages_count
+    attributes :item_ids, :cancellation_reason, :unread_messages_count,
+      :user_submitted_order_count, :user_awaiting_dispatch_order_count, :user_cancelled_order_count, :user_closed_order_count
+
```
Comment (abhinav098) [inline app/serializers/api/v1/order_serializer.rb]: Please take care of no of chars in line. They should not exceed 80 chars.

File: app/serializers/api/v1/order_serializer.rb [hunk 2]
```diff
@@ -33,6 +35,37 @@ def unread_messages_count__sql
       "(select count(*) from subscriptions s where s.order_id = orders.id and s.state = 'unread' and s.user_id = orders.created_by_id)"
     end
 
+    def user_submitted_order_count
+      Order.where(state: 'submitted', created_by_id: object.created_by_id).count
+    end
+
+    def user_submitted_order_count__sql
+      "(select count(*) from orders where orders.state = 'submitted' and orders.created_by_id = orders.created_by_id)"
```
Comment (abhinav098) [inline app/serializers/api/v1/order_serializer.rb]: Please take care length of chars here to here too.

File: app/serializers/api/v1/order_serializer.rb [hunk 3]
```diff
@@ -33,6 +35,37 @@ def unread_messages_count__sql
...
+    end
+
+    def user_awaiting_dispatch_order_count
+      Order.where(state: 'awaiting_dispatch', created_by_id: object.created_by_id).count
+    end
+
+    def user_awaiting_dispatch_order_count__sql
+      "(select count(*) from orders where orders.state = 'awaiting_dispatch' and orders.created_by_id = orders.created_by_id)"
```
Comment (abhinav098) [inline app/serializers/api/v1/order_serializer.rb]: ^ Please take care length of chars here to here too.

File: app/serializers/api/v1/order_serializer.rb [hunk 4]
```diff
@@ -33,6 +35,37 @@ def unread_messages_count__sql
...
+    def user_submitted_order_count
+      Order.where(state: 'submitted', created_by_id: object.created_by_id).count
+    end
+
+    def user_submitted_order_count__sql
+      "(select count(*) from orders where orders.state = 'submitted' and orders.created_by_id = orders.created_by_id)"
+    end
+
+    def user_awaiting_dispatch_order_count
+      Order.where(state: 'awaiting_dispatch', created_by_id: object.created_by_id).count
```
Comment (namrataukirde) [inline app/serializers/api/v1/order_serializer.rb]: Can we add generalize scope here and pass a parameter as state?

File: spec/serializers/api/v1/order_serializer_spec.rb [hunk 5]
```diff
@@ -1,13 +1,18 @@
 require 'rails_helper'
 
 describe Api::V1::OrderSerializer do
+  TOTAL_REQUESTS_STATES = ["submitted", "awaiting_dispatch", "closed", "cancelled"]
   let(:country) { create :country }
   let(:user) { create :user }
+  TOTAL_REQUESTS_STATES.each do |state|
+    let!(:"#{state}_order") { create :order, :with_orders_packages, :"with_state_#{state}", created_by_id: user.id, status: nil }
+  end
```
Comment (namrataukirde) [inline spec/serializers/api/v1/order_serializer_spec.rb]: 👍

File: spec/serializers/api/v1/order_serializer_spec.rb [hunk 6]
```diff
@@ -1,13 +1,18 @@
 require 'rails_helper'
 
 describe Api::V1::OrderSerializer do
+  TOTAL_REQUESTS_STATES = ["submitted", "awaiting_dispatch", "closed", "cancelled"]
```
Comment (namrataukirde) [inline spec/serializers/api/v1/order_serializer_spec.rb]: Add `.freeze` on constant.

File: app/serializers/api/v1/order_serializer.rb [hunk 7]
```diff
@@ -33,6 +35,41 @@ def unread_messages_count__sql
       "(select count(*) from subscriptions s where s.order_id = orders.id and s.state = 'unread' and s.user_id = orders.created_by_id)"
     end
 
+    def user_submitted_order_count
+      Order.users_order_count(object.created_by_id, 'submitted')
```
Comment (Darshan-Chauhan) [inline app/serializers/api/v1/order_serializer.rb]: Is it ok to request data from serializer? 
I think serializer should only be used for serializing data. 

CC: @namrataukirde

**Pull request closed without merging**